connection.go

```diff
@@ -993,6 +999,11 @@ func (conn *Connection) nextRequestId() (requestId uint32) {
 // An error is returned if the request was formed incorrectly, or failed to
 // create the future.
 func (conn *Connection) Do(req Request) *Future {
+	if stickyReq, ok := req.(ConnectedObject); ok {
+		if stickyReq.Conn() != conn {
+			return &Future{err: ErrStrangerConn}
```

Comment:
A single way to create an object simplifies life in the future:

NewFuture() + fut.SetError()

Reply:
Fixed.